middle_end/flambda2/from_lambda/lambda_conversions.ml

```diff
@@ -185,6 +185,7 @@ let convert_init_or_assign (i_or_a : L.initialization_or_assignment) :
     P.Init_or_assign.t =
   match i_or_a with
   | Assignment -> Assignment
+  | Local_assignment -> assert false (* temporary *)
```

Comment:
After merge! I split the larger local allocs branch into separate backend and middle-end patches to make the PRs here smaller. This one, the backend patch, changes `Lambda.initialization_or_assignment` (because of how it's used in Cmm), and so I made the minimal corresponding change to the middle end to make CI pass. This will be removed in the next PR, which includes the full middle-end changes.